oci/table_oci_containerengine_cluster.go

```diff
@@ -0,0 +1,284 @@
+	logger.Debug("listContainerEngineClusters", "Compartment", compartment, "OCI_REGION", region)
+
+	equalQuals := d.KeyColumnQuals
+
+	// Return nil, if given compartment_id doesn't match
+	if equalQuals["compartment_id"] != nil && compartment != equalQuals["compartment_id"].GetStringValue() {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := containerEngineService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build request parameters
+	request, isValid := buildContainerEngineClusterFilters(equalQuals, logger)
+	if !isValid {
+		return nil, nil
+	}
+	request.CompartmentId = types.String(compartment)
+	request.Limit = types.Int(1000)
+	request.RequestMetadata = common.RequestMetadata{
+		RetryPolicy: getDefaultRetryPolicy(),
+	}
+
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit < int64(*request.Limit) {
+			request.Limit = types.Int(int(*limit))
+		}
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.ContainerEngineClient.ListClusters(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, cluster := range response.Items {
+			d.StreamListItem(ctx, cluster)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		if response.OpcNextPage != nil {
+			request.Page = response.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+	return nil, err
+}
+
+//// HYDRATE FUNCTION
+
+func getContainerEngineCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("getContainerEngineClusters", "Compartment", compartment, "OCI_REGION", region)
+
+	// Restrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	var id string
+	if h.Item != nil {
+		id = *h.Item.(containerengine.ClusterSummary).Id
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+	}
+
+	// handle empty id in get call
+	if id == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := containerEngineService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	request := containerengine.GetClusterRequest{
+		ClusterId: types.String(id),
+		RequestMetadata: common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(),
+		},
+	}
+
+	response, err := session.ContainerEngineClient.GetCluster(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	return response.Cluster, nil
+}
+
+// Build additional filters
+func buildContainerEngineClusterFilters(equalQuals plugin.KeyColumnEqualsQualMap, logger hclog.Logger) (containerengine.ListClustersRequest, bool) {
+	request := containerengine.ListClustersRequest{}
+	isValid := true
+	if equalQuals["name"] != nil {
```

Comment:
Add extra check for empty value check, If you will try the below query then you will get the InvalidParameterException.

`elect * from oci_containerengine_cluster where name = ' '` or `elect * from oci_containerengine_cluster where name = '<more than 1 empty space>' ` or `elect * from oci_containerengine_cluster where name = ''`